Incompatibility with sodium-extra

**Describe the bug**
Minecraft's main menu can't load up because of infinite loading loop when using Sodium Extra 0.3.9 mod

**To Reproduce**
1. Download and install Sodium and Sodium Extra aswell
2. Install Blanket
3. Start the client via launcher
4. Main menu can't load up, keeping reloading in loop

**Expected behavior**
Minecraft starts without loop and errors


**Desktop (please complete the following information):**
 - Minecraft 1.18.1 latest fabric
 - Latest snapshot of blanket
 - latest versions of Sodium and Sodium extra released on curseforge

**Minecraft crash report**
Found these records in log

```
Caused by: java.lang.BootstrapMethodError: java.lang.RuntimeException: Mixin transformation of net.minecraft.class_907 failed
	at net.minecraft.class_5619.<clinit>(class_5619.java:50) ~[client-intermediary.jar:?]
	at net.minecraft.class_898.method_14491(class_898.java:369) ~[client-intermediary.jar:?]
	at net.minecraft.class_4013.method_29490(class_4013.java:15) ~[client-intermediary.jar:?]
	at java.util.concurrent.CompletableFuture$UniRun.tryFire(CompletableFuture.java:787) ~[?:?]
	... 113 more
Caused by: java.lang.RuntimeException: Mixin transformation of net.minecraft.class_907 failed
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.getPostMixinClassByteArray(KnotClassDelegate.java:252) ~[fabric-loader-0.12.12.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.tryLoadClass(KnotClassDelegate.java:150) ~[fabric-loader-0.12.12.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:155) ~[fabric-loader-0.12.12.jar:?]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:520) ~[?:?]
	at java.lang.ClassLoader.defineClass1(Native Method) ~[?:?]
	at java.lang.ClassLoader.defineClass(ClassLoader.java:1012) ~[?:?]
	at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:150) ~[?:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassLoader.defineClassFwd(KnotClassLoader.java:218) ~[fabric-loader-0.12.12.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.tryLoadClass(KnotClassDelegate.java:174) ~[fabric-loader-0.12.12.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:155) ~[fabric-loader-0.12.12.jar:?]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:520) ~[?:?]
	at net.minecraft.class_5619.<clinit>(class_5619.java:50) ~[client-intermediary.jar:?]
	at net.minecraft.class_898.method_14491(class_898.java:369) ~[client-intermediary.jar:?]
	at net.minecraft.class_4013.method_29490(class_4013.java:15) ~[client-intermediary.jar:?]
	at java.util.concurrent.CompletableFuture$UniRun.tryFire(CompletableFuture.java:787) ~[?:?]
	... 113 more
Caused by: org.spongepowered.asm.mixin.transformer.throwables.MixinTransformerError: An unexpected critical error was encountered
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:392) ~[sponge-mixin-0.10.7+mixin.0.8.4.jar:0.10.7+mixin.0.8.4]
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClass(MixinTransformer.java:234) ~[sponge-mixin-0.10.7+mixin.0.8.4.jar:0.10.7+mixin.0.8.4]
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClassBytes(MixinTransformer.java:202) ~[sponge-mixin-0.10.7+mixin.0.8.4.jar:0.10.7+mixin.0.8.4]
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.getPostMixinClassByteArray(KnotClassDelegate.java:247) ~[fabric-loader-0.12.12.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.tryLoadClass(KnotClassDelegate.java:150) ~[fabric-loader-0.12.12.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:155) ~[fabric-loader-0.12.12.jar:?]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:520) ~[?:?]
	at java.lang.ClassLoader.defineClass1(Native Method) ~[?:?]
	at java.lang.ClassLoader.defineClass(ClassLoader.java:1012) ~[?:?]
	at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:150) ~[?:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassLoader.defineClassFwd(KnotClassLoader.java:218) ~[fabric-loader-0.12.12.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.tryLoadClass(KnotClassDelegate.java:174) ~[fabric-loader-0.12.12.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:155) ~[fabric-loader-0.12.12.jar:?]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:520) ~[?:?]
	at net.minecraft.class_5619.<clinit>(class_5619.java:50) ~[client-intermediary.jar:?]
	at net.minecraft.class_898.method_14491(class_898.java:369) ~[client-intermediary.jar:?]
	at net.minecraft.class_4013.method_29490(class_4013.java:15) ~[client-intermediary.jar:?]
	at java.util.concurrent.CompletableFuture$UniRun.tryFire(CompletableFuture.java:787) ~[?:?]
	... 113 more
Caused by: org.spongepowered.asm.mixin.injection.throwables.InjectionError: Critical injection failure: Redirector useTimeOfDay(Lnet/minecraft/class_1937;)J in sodium-extra.mixins.json:entity.MixinGuardianEntityRenderer from mod sodium-extra failed injection check, (0/1) succeeded. Scanned 1 target(s). Using refmap sodium-extra-mc1.18.1-refmap.json
	at org.spongepowered.asm.mixin.injection.struct.InjectionInfo.postInject(InjectionInfo.java:468) ~[sponge-mixin-0.10.7+mixin.0.8.4.jar:0.10.7+mixin.0.8.4]
	at org.spongepowered.asm.mixin.transformer.MixinTargetContext.applyInjections(MixinTargetContext.java:1381) ~[sponge-mixin-0.10.7+mixin.0.8.4.jar:0.10.7+mixin.0.8.4]
	at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.applyInjections(MixinApplicatorStandard.java:1052) ~[sponge-mixin-0.10.7+mixin.0.8.4.jar:0.10.7+mixin.0.8.4]
	at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.applyMixin(MixinApplicatorStandard.java:400) ~[sponge-mixin-0.10.7+mixin.0.8.4.jar:0.10.7+mixin.0.8.4]
	at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.apply(MixinApplicatorStandard.java:325) ~[sponge-mixin-0.10.7+mixin.0.8.4.jar:0.10.7+mixin.0.8.4]
	at org.spongepowered.asm.mixin.transformer.TargetClassContext.apply(TargetClassContext.java:421) ~[sponge-mixin-0.10.7+mixin.0.8.4.jar:0.10.7+mixin.0.8.4]
	at org.spongepowered.asm.mixin.transformer.TargetClassContext.applyMixins(TargetClassContext.java:403) ~[sponge-mixin-0.10.7+mixin.0.8.4.jar:0.10.7+mixin.0.8.4]
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:363) ~[sponge-mixin-0.10.7+mixin.0.8.4.jar:0.10.7+mixin.0.8.4]
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClass(MixinTransformer.java:234) ~[sponge-mixin-0.10.7+mixin.0.8.4.jar:0.10.7+mixin.0.8.4]
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClassBytes(MixinTransformer.java:202) ~[sponge-mixin-0.10.7+mixin.0.8.4.jar:0.10.7+mixin.0.8.4]
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.getPostMixinClassByteArray(KnotClassDelegate.java:247) ~[fabric-loader-0.12.12.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.tryLoadClass(KnotClassDelegate.java:150) ~[fabric-loader-0.12.12.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:155) ~[fabric-loader-0.12.12.jar:?]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:520) ~[?:?]
	at java.lang.ClassLoader.defineClass1(Native Method) ~[?:?]
	at java.lang.ClassLoader.defineClass(ClassLoader.java:1012) ~[?:?]
	at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:150) ~[?:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassLoader.defineClassFwd(KnotClassLoader.java:218) ~[fabric-loader-0.12.12.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.tryLoadClass(KnotClassDelegate.java:174) ~[fabric-loader-0.12.12.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:155) ~[fabric-loader-0.12.12.jar:?]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:520) ~[?:?]
	at net.minecraft.class_5619.<clinit>(class_5619.java:50) ~[client-intermediary.jar:?]
	at net.minecraft.class_898.method_14491(class_898.java:369) ~[client-intermediary.jar:?]
	at net.minecraft.class_4013.method_29490(class_4013.java:15) ~[client-intermediary.jar:?]
	at java.util.concurrent.CompletableFuture$UniRun.tryFire(CompletableFuture.java:787) ~[?:?]
	... 113 more
```

--- a/src/main/java/io/github/blanketmc/blanket/ClientFixesPreLaunch.java
+++ b/src/main/java/io/github/blanketmc/blanket/ClientFixesPreLaunch.java
@@ -1,8 +1,10 @@
 package io.github.blanketmc.blanket;
 
+import net.fabricmc.loader.api.FabricLoader;
 import net.fabricmc.loader.api.entrypoint.PreLaunchEntrypoint;
 import net.fabricmc.loader.impl.util.UrlConversionException;
 import net.fabricmc.loader.impl.util.UrlUtil;
+import org.spongepowered.asm.mixin.Mixins;
 
 import java.lang.reflect.InvocationTargetException;
 import java.lang.reflect.Method;
@@ -10,6 +12,16 @@ import java.net.URL;
 
 public class ClientFixesPreLaunch implements PreLaunchEntrypoint {
     // Allow the mod to mixin directly into library classes
+
+    /**
+     * load mixins with conditions
+     */
+    private static void loadMixinsSelectively() {
+        if (!FabricLoader.getInstance().isModLoaded("sodium-extra")) {
+            Mixins.addConfiguration("blanket-client-tweaks_no-sodium-extra.json");
+        }
+    }
+
 
     private static final String[] mixinTargets = {
             "com/mojang/authlib/yggdrasil/YggdrasilUserApiService.class"
@@ -28,6 +40,8 @@ public class ClientFixesPreLaunch implements PreLaunchEntrypoint {
 
     @Override
     public void onPreLaunch() {
+        loadMixinsSelectively();
+
         try {
             ClassLoader classLoader = getClass().getClassLoader();
             Method method = classLoader.getClass().getMethod("addURL", URL.class);

--- a/src/main/java/io/github/blanketmc/blanket/Config.java
+++ b/src/main/java/io/github/blanketmc/blanket/Config.java
@@ -48,7 +48,7 @@ public final class Config {
 
     //by FX - PR0CESS
     @ConfigEntry(
-            description = "Fix guardian beam not rendering due to world time being too high",
+            description = "Fix guardian beam not rendering due to world time being too high\nSodium-extra has the same fix",
             issues = "MC-165595",
             categories = {BUGFIX, RECOMMENDED, RENDER}
     )

--- a/src/main/resources/blanket-client-tweaks.mixins.json
+++ b/src/main/resources/blanket-client-tweaks.mixins.json
@@ -10,7 +10,6 @@
     "fixes.Entity_lavaDamageMixin",
     "fixes.Entity_localIdMixin",
     "fixes.FireworkRocketEntity_particleMixin",
-    "fixes.GuardianEntityRenderer_beamMixin",
     "fixes.HandledScreen_mouseCloseMixin",
     "fixes.WolfEntity_tailSpinMixin",
     "fixes.YggdrasilUserApiService_syncBlockListMixin",

--- a/src/main/resources/blanket-client-tweaks_no-sodium-extra.json
+++ b/src/main/resources/blanket-client-tweaks_no-sodium-extra.json
@@ -1,0 +1,12 @@
+{
+  "required": true,
+  "minVersion": "0.8",
+  "package": "io.github.blanketmc.blanket.mixin",
+  "compatibilityLevel": "JAVA_16",
+  "client": [
+    "fixes.GuardianEntityRenderer_beamMixin"
+  ],
+  "injectors": {
+    "defaultRequire": 0
+  }
+}

